File: DSA/python/queue.py
#!/usr/bin/python3

from stack import Error 
class Queue:
    def __init__(self):
        self.Max_size=int(input("enter the size of your queue "))
        self.items=[]
        self.front=-1
        self.rear=-1
    def Enqueue(self,ele):
        if len(self.items)>=self.Max_size:
            raise Error.OverFlowError
        else:
            if self.front==-1 and self.rear==-1:
                self.items.append(ele)
                self.front=0
                self.rear=0
                print(f"{ele} added to queue")
                return 0
            else:
                self.items.insert(self.rear,ele)
                self.front += 1
                print(f"{ele} added to queue")
                return 0
    def Display(self):
        print("the elements of the queue are:")
        for i in self.items:
            print(i)
    def Dequeue(self):
        if self.front==-1:
            raise Error.UnderFlowerror
        else:
            print(self.items[self.front],"removed from queue")
            self.items.remove(self.items[self.front])
            self.front -= 1
            return 0

if __name__ == '__main__':
    Q=Queue()
    while True:
        print("press 1 to enqueue\npress 2 to dequeue\npress 3 to display\n")
        choice=int(input())
        if choice==1:
            Q.Enqueue(input("enter Element to enqueue: "))
        elif choice == 2:
            Q.Dequeue()
        elif choice == 3:
            Q.Display()
        else:
            print("invalid choice exiting")
            break

File: cpp/queue.cpp
# include <iostream>
# define Max_size 5

using namespace std;

class Linear_Queue{
        int queue[Max_size];
        int front=-1;
        int rear=-1;
        public:
                int Enqueue(){
                        int ele;
                        if (rear == Max_size-1){
                                cout<<"OverflowError: Queue is full\n\n";
                                return 1;
                        }
                        else {
                                cout<<"enter element to enqueue:\n";
                                cin>>ele;
                                if (front==-1 && rear==-1)
                                        front++;
                                else
                                        ;
                                rear++;
                                queue[rear]=ele;
                                cout<<ele<<" Added to queue\n";
                                return 0;
                        }
                }
                int Display(){
                        if (front==-1){
                                cout<<"Queue is empty\n";
                                return 1;
                        }
                        else{
                                cout<<"The elements of the queue are\n";
                                for(int i = front; i <=rear; i++){
                                        cout<<queue[i]<<" ";
                                }
                                return 0;
                        }
                }

                int Dequeue(){
                        if (front>rear){
                                cout<<"UnderFlowerror: Queue is empty\n";
                                return 1;
                        }
                        else{
                                cout<<queue[front]<<" Removed from queue\n";
                                front++;
                                return 0;
                        }
                }
	};


int main(){
        Linear_Queue Q;
        int choice;
        int loop=3;
        while(loop==3){
                cout<<"\npress 1 to enqueue \npress 2 to dequeue\npress 3 to display\n";
                cin>>choice;
                if (choice==1){
                        if(Q.Enqueue())
                                break;
                        else
                                choice=0;
                }
                else if (choice==2){
                        if (Q.Dequeue())
                                break;
                        else
                                choice;
                }
                else if(choice == 3) {
                        if (Q.Display())
                                break;
                        else
                                choice;
                }
                else{

                        cout<<"Invalid choice exiting";
                        break;
                }
        }
}
